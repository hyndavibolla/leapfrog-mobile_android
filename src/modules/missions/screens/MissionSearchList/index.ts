export { default as MissionSearchList } from './MissionSearchList';
