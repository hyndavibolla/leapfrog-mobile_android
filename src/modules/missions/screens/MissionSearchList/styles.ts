import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE, MEASURE } from '_constants';

export const styles = {
  container: css`
    padding-top: ${MEASURE.MODAL_PADDING}px;
    padding-horizontal: 16px;
    flex: 1;
  `,
  brandsContainer: css`
    margin-top: 24px;
  `,
  brandContainer: css`
    background-color: ${COLOR.WHITE};
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    margin-bottom: 8px;
    padding: 16px;
  `,
  brandLogo: css`
    margin-right: 16px;
  `,
  brandName: css`
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
    color: ${COLOR.BLACK};
  `,
  emptyStateContainer: css`
    flex: 1;
    justify-content: space-around;
    padding: 32px;
  `,
  emptyStateMessageContainer: css`
    justify-self: center;
    align-items: center;
    padding: 32px;
  `,
  emptyStateTitle: css`
    margin-top: 20px;
    text-align: center;
  `,
  emptyStateText: css`
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.SMALL};
    text-align: center;
    margin-top: 20px;
  `
};
