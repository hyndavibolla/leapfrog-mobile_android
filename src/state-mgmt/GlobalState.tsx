import React, { createContext, memo, PropsWithChildren, useReducer, Dispatch, useEffect } from 'react';

import { IGlobalState, Deps, IReducer, IGlobalContext } from '_models/general';
import { reducer as coreReducer } from './core/reducer';
import { reducer as userReducer } from './user/reducer';
import { reducer as gameReducer } from './game/reducer';
import { reducer as missionReducer } from './mission/reducer';
import { reducer as rewardReducer } from './reward/reducer';
import { reducer as surveyReducer } from './survey/reducer';
import { reducer as messagesReducer } from './messages/reducer';
import { reducer as cardLinkReducer } from './cardLink/reducer';
import { reducer as giftCardReducer } from './giftCard/reducer';
import { initialState as coreInitialState } from './core/state';
import { initialState as userInitialState } from './user/state';
import { initialState as gameInitialState } from './game/state';
import { initialState as missionInitialState } from './mission/state';
import { initialState as rewardInitialState } from './reward/state';
import { initialState as surveyInitialState } from './survey/state';
import { initialState as messagesInitialState } from './messages/state';
import { initialState as cardLinkInitialState } from './cardLink/state';
import { initialState as giftCardInitialState } from './giftCard/state';
import { combineReducers } from '_utils/combineReducers';
import { GeneralModel } from '_models';
import { deepClone } from '_utils/deepClone';

export const getInitialState = () =>
  deepClone<IGlobalState>({
    core: coreInitialState,
    user: userInitialState,
    game: gameInitialState,
    mission: missionInitialState,
    reward: rewardInitialState,
    survey: surveyInitialState,
    messages: messagesInitialState,
    cardLink: cardLinkInitialState,
    giftCard: giftCardInitialState
  });

export const combinedReducer = combineReducers<GeneralModel.IGlobalState>({
  core: coreReducer,
  user: userReducer,
  game: gameReducer,
  mission: missionReducer,
  reward: rewardReducer,
  survey: surveyReducer,
  messages: messagesReducer,
  cardLink: cardLinkReducer,
  giftCard: giftCardReducer
});

export const GlobalContext = createContext<{ state: IGlobalState; dispatch: Dispatch<any>; deps: Deps }>({
  state: getInitialState(),
  dispatch: /* istanbul ignore next */ () => null,
  deps: null
} as IGlobalContext);

export const GlobalProvider = memo(
  ({
    children,
    deps,
    initState,
    combinedReducers
  }: PropsWithChildren<{
    deps: Deps;
    initState: IGlobalState;
    combinedReducers: IReducer<IGlobalState>;
  }>) => {
    const [state, dispatch] = useReducer(combinedReducers, initState);
    useEffect(() => deps.stateSnapshot.set(state), [state]); // eslint-disable-line react-hooks/exhaustive-deps
    return <GlobalContext.Provider value={{ state, dispatch, deps } as IGlobalContext}>{children}</GlobalContext.Provider>;
  }
);
