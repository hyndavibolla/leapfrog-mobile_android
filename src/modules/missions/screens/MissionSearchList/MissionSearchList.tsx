import React, { memo, useCallback, useContext, useEffect, useMemo, useState } from 'react';
import { ScrollView, View, Text, TouchableHighlight } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';

import { SearchInput } from '_components/SearchInput';
import { Card } from '_components/Card';
import { BrandLogo } from '_components/BrandLogo';
import { Title, TitleType } from '_components/Title';
import { Icon } from '_commons/components/atoms/Icon';
import { StreakListSkeleton } from './components/MissionSearchListSkeleton';

import { useTestingHelper } from '_utils/useTestingHelper';
import { useEventTracker } from '_state_mgmt/core/hooks';
import { useMissionFreeSearch } from '_state_mgmt/mission/hooks';
import { KnownMissionSearchKey } from '_state_mgmt/mission/state';
import { actions as missionActions } from '_state_mgmt/mission/actions';
import { GlobalContext } from '_state_mgmt/GlobalState';
import { MissionListType } from '_models/mission';
import { MissionModel } from '_models';

import { normalizeTextForSearching } from '_utils/normalizeTextForSearching';
import { COLOR, ENV, ForterActionType, TealiumEventType, UxObject, ICON, ROUTES } from '_constants';
import NoResults from '_assets/shared/brandNoResults.svg';

import { styles } from './styles';

export interface Props {
  route: {
    params: {
      streakId: string;
    };
  };
  navigation: StackNavigationProp<any>;
}

function StreakList({ navigation, route }: Props) {
  const [text, setText] = useState('');
  const { trackUserEvent } = useEventTracker();

  const { getTestIdProps } = useTestingHelper('streak-list');
  const [onSearch, isSearching = true] = useMissionFreeSearch();

  const {
    state: {
      mission: { missionSearchMap, missionMap }
    },
    dispatch
  } = useContext(GlobalContext);

  useEffect(() => {
    dispatch(missionActions.flushSearchList(KnownMissionSearchKey.STREAK_BRANDS));
    onSearch(
      KnownMissionSearchKey.STREAK_BRANDS,
      MissionListType.DEFAULT,
      undefined,
      undefined,
      ENV.MISSION_LIMIT.FULL,
      undefined,
      undefined,
      route.params.streakId
    );
  }, [dispatch, onSearch, route.params.streakId]);

  const missionList = useMemo(
    () =>
      (missionSearchMap[KnownMissionSearchKey.STREAK_BRANDS] || /* istanbul ignore next */ [])
        .map(uuid => missionMap[uuid])
        .filter(brand => {
          const normalizedBrand = normalizeTextForSearching(brand?.brandName);
          return normalizedBrand && normalizedBrand.startsWith(normalizeTextForSearching(text));
        })
        .sort((a, b) => a.brandName.localeCompare(b.brandName)),
    [missionSearchMap, missionMap, text]
  );

  const ListEmptyComponent = useCallback(() => {
    return (
      <View style={styles.emptyStateContainer} {...getTestIdProps('empty-state')}>
        <View style={styles.emptyStateMessageContainer}>
          <NoResults />
          <Title style={styles.emptyStateTitle} type={TitleType.HEADER}>
            {'Woops! No search results :('}
          </Title>
        </View>
      </View>
    );
  }, [getTestIdProps]);

  const handleMissionPress = useCallback(
    (mission: MissionModel.IMission) => {
      trackUserEvent(
        TealiumEventType.SELECT_MISSION,
        {
          event_type: mission.brandName,
          event_detail: mission.offerId,
          uxObject: UxObject.LIST,
          brand_name: mission.brandName,
          brand_id: mission.brandRequestorId,
          brand_category: mission.brandCategories[0]
        },
        ForterActionType.TAP
      );
      navigation.navigate(ROUTES.MISSION_DETAIL, {
        brandRequestorId: mission.brandRequestorId,
        isAvailableStreakIndicator: true
      });
    },
    [navigation, trackUserEvent]
  );

  if (isSearching) return <StreakListSkeleton />;

  return (
    <View style={styles.container} {...getTestIdProps('container')}>
      <SearchInput
        placeholder="Search brands"
        onChange={setText}
        value={text}
        hideFilters
        placeholderTextColor={COLOR.WHITE}
        Icon={<Icon name={ICON.SEARCH} color={COLOR.WHITE} />}
      />
      {!missionList.length ? (
        <ListEmptyComponent />
      ) : (
        <ScrollView style={styles.brandsContainer}>
          {missionList.map(brand => (
            <TouchableHighlight
              underlayColor="transparent"
              onPress={() => handleMissionPress(brand)}
              {...getTestIdProps('brand-container-btn')}
              key={brand.uuid}
            >
              <Card style={styles.brandContainer}>
                <BrandLogo image={brand?.brandLogo} category={brand?.pointsAwarded?.conditions[0]?.category} style={styles.brandLogo} streakIndicator />
                <Text style={styles.brandName}>{brand.brandName}</Text>
              </Card>
            </TouchableHighlight>
          ))}
        </ScrollView>
      )}
    </View>
  );
}
export default memo(StreakList);
