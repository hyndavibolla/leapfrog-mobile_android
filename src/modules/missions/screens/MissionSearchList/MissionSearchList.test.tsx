import React from 'react';
import { fireEvent, waitFor, waitForElementToBeRemoved } from '@testing-library/react-native';

import MissionSearchList, { Props } from './MissionSearchList';

import { getInitialState } from '_state_mgmt/GlobalState';
import { GeneralModel } from '_models';
import { Deps } from '_models/general';

import { getMockDeps } from '_test_utils/getMockDeps';
import { getMission_1 } from '_test_utils/entities';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { ROUTES } from '_constants/routes';

describe('MissionSearchList', () => {
  let state: GeneralModel.IGlobalState;
  let deps: Deps;
  let props: Props;

  beforeEach(() => {
    state = getInitialState();
    deps = getMockDeps();
    props = {
      navigation: {
        navigate: jest.fn()
      } as any,
      route: {
        params: {
          streakId: 'streak-1'
        }
      }
    };
  });

  it('should render', async () => {
    deps.apiService.fetchMissionList = jest.fn().mockResolvedValueOnce({
      userId: 'buttonId',
      missions: Array.from(new Array(60)).map((_, index) => ({
        ...getMission_1(),
        uuid: String(index)
      }))
    });
    const { toJSON } = renderWithGlobalContext(<MissionSearchList {...props} />, deps, state);
    await waitFor(() => expect(toJSON()).toMatchSnapshot());
  });

  it('should render the skeleton while the data is fetching and then remove it', async () => {
    deps.apiService.fetchMissionList = jest.fn().mockResolvedValueOnce({
      userId: 'buttonId',
      missions: Array.from(new Array(60)).map((_, index) => ({
        ...getMission_1(),
        uuid: String(index)
      }))
    });
    const { getByTestId } = renderWithGlobalContext(<MissionSearchList {...props} />, deps, state);
    await waitForElementToBeRemoved(() => getByTestId('streak-list-skeleton-container'));
  });

  it('should render a empty state when there are no missions', async () => {
    deps.apiService.fetchMissionList = jest.fn().mockResolvedValueOnce({
      userId: 'buttonId',
      missions: []
    });
    const { getByTestId } = renderWithGlobalContext(<MissionSearchList {...props} />, deps, state);
    await waitFor(() => expect(getByTestId('streak-list-empty-state')).toBeTruthy());
  });

  it('should navigate to mission detail when a mission was pressed', async () => {
    deps.apiService.fetchMissionList = jest.fn().mockResolvedValueOnce({
      userId: 'buttonId',
      missions: Array.from(new Array(60)).map((_, index) => ({
        ...getMission_1(),
        uuid: String(index)
      }))
    });

    const { findAllByTestId } = renderWithGlobalContext(<MissionSearchList {...props} />, deps, state);

    fireEvent.press((await findAllByTestId('streak-list-brand-container-btn'))[0]);
    expect(props.navigation.navigate).toBeCalledWith(ROUTES.MISSION_DETAIL, {
      brandRequestorId: getMission_1().brandRequestorId,
      isAvailableStreakIndicator: true
    });
  });
});
