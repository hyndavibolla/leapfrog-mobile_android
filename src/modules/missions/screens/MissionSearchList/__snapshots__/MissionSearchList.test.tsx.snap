// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionSearchList should render 1`] = `
<RCTScrollView
  accessibilityLabel="streak-list-skeleton-container"
  accessible={true}
  contentContainerStyle={
    Object {
      "paddingBottom": 32,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 32,
    }
  }
  testID="streak-list-skeleton-container"
>
  <View>
    <div
      animation={
        Object {
          "0": Object {
            "opacity": 1,
          },
          "0.5": Object {
            "opacity": 0.35,
          },
          "1": Object {
            "opacity": 1,
          },
        }
      }
      duration={2000}
      easing="ease"
      iterationCount="infinite"
    >
      <View
        color="transparent"
        dir="row"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": "auto",
              "width": "100%",
            },
            undefined,
          ]
        }
        width="100%"
      >
        <View
          flex={1}
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "column",
                "height": 40,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          height="40px"
          radius={40}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 40,
                "borderBottomRightRadius": 40,
                "borderTopLeftRadius": 40,
                "borderTopRightRadius": 40,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginLeft": 15,
              },
            ]
          }
          width="40px"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
      <View
        color="#EBEDF2"
        dir="row"
        height="80px"
        style={
          Array [
            Object {
              "backgroundColor": "#EBEDF2",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "height": 80,
              "width": "100%",
            },
            Object {
              "alignItems": "center",
              "padding": 15,
            },
          ]
        }
      >
        <View
          height="40px"
          radius={20}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexBasis": "auto",
                "flexDirection": "column",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 40,
                "width": 40,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          width="40px"
        />
        <View
          dir="row"
          height="20px"
          radius={10}
          style={
            Array [
              Object {
                "backgroundColor": "#DFE3EC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": 20,
                "width": "60%",
              },
              undefined,
            ]
          }
          width="60%"
        />
      </View>
      <View
        style={
          Object {
            "height": 15,
            "width": "100%",
          }
        }
      />
    </div>
  </View>
</RCTScrollView>
`;
